api/proto/teleport/summarizer/v1/summarizer.proto

```diff
@@ -107,3 +109,35 @@ message InferencePolicySpec {
   // match the list of kinds will be summarized using this model.
   string filter = 3;
 }
+
+// SummaryState is the state of the summarization process.
+enum SummaryState {
+  SUMMARY_STATE_UNSPECIFIED = 0;
+  SUMMARY_STATE_PENDING = 1;
+  SUMMARY_STATE_SUCCESS = 2;
+  SUMMARY_STATE_ERROR = 3;
+}
+
+// Summary represents a summary of a session recording. This format is used to
+// store the summaries in the session storage and return it with gRPC.
+message Summary {
+  // sessionId is an ID of the session whose recording got summarized.
+  string session_id = 1;
+  // State is the state of the summarization process.
+  SummaryState state = 2;
+  // InferenceStartedAt is the time when the summarization process started.
+  google.protobuf.Timestamp inference_started_at = 3;
+  // InferenceFinishedAt is the time when the summarization process finished.
+  google.protobuf.Timestamp inference_finished_at = 4;
+  // Content is the main text content of the summary. Available if the state is
+  // SUMMARY_STATE_SUCCESS.
+  string content = 5;
+  // ModelName is the name of the `InferenceModel` resource that was used to
+  // generate this summary.
+  string model_name = 6;
+  // SessionEndEvent is the event that ended the summarized session.
+  events.OneOf session_end_event = 7;
+  // ErrorMessage is an error message if the summarization failed. Available if
+  // the state is SUMMARY_STATE_ERROR.
+  string error_message = 8;
```

Comment:
Is this a common pattern? Wouldn't you use a gRPC error for this?

Reply:
@juliaogris It's not a common pattern, but this object will be going to be stored on s3, and then returned via another call. In that another call we could technically return an error if we see an error message here, but I'd like to separate these anyway to make it easier to distinguish between a successful call that returns a state of a summary that has previously been unsuccessfully generated and an unsuccessful call that doesn't return any useful data.

Reply:
(In other words: think about it as a resource whose spec contains an information about an event from the past; in this case, the event is summary inference.)